File: SDKs/googleapis.com/recommender/v1beta1/go/pkg/models/operations/recommenderprojectslocationsrecommendersupdateconfig.go
package operations

import (
"openapi/pkg/models/shared")

type RecommenderProjectsLocationsRecommendersUpdateConfigPathParams struct {
    Name string `pathParam:"style=simple,explode=false,name=name"`
    
}

type RecommenderProjectsLocationsRecommendersUpdateConfigQueryParams struct {
    DollarXgafv *shared.XgafvEnum `queryParam:"style=form,explode=true,name=$.xgafv"`
    AccessToken *string `queryParam:"style=form,explode=true,name=access_token"`
    Alt *shared.AltEnum `queryParam:"style=form,explode=true,name=alt"`
    Callback *string `queryParam:"style=form,explode=true,name=callback"`
    Fields *string `queryParam:"style=form,explode=true,name=fields"`
    Key *string `queryParam:"style=form,explode=true,name=key"`
    OauthToken *string `queryParam:"style=form,explode=true,name=oauth_token"`
    PrettyPrint *bool `queryParam:"style=form,explode=true,name=prettyPrint"`
    QuotaUser *string `queryParam:"style=form,explode=true,name=quotaUser"`
    UpdateMask *string `queryParam:"style=form,explode=true,name=updateMask"`
    UploadType *string `queryParam:"style=form,explode=true,name=uploadType"`
    UploadProtocol *string `queryParam:"style=form,explode=true,name=upload_protocol"`
    ValidateOnly *bool `queryParam:"style=form,explode=true,name=validateOnly"`
    
}

type RecommenderProjectsLocationsRecommendersUpdateConfigSecurity struct {
    Oauth2 shared.SchemeOauth2 `security:"scheme,type=oauth2"`
    Oauth2c shared.SchemeOauth2c `security:"scheme,type=oauth2"`
    
}

type RecommenderProjectsLocationsRecommendersUpdateConfigRequest struct {
    PathParams RecommenderProjectsLocationsRecommendersUpdateConfigPathParams 
    QueryParams RecommenderProjectsLocationsRecommendersUpdateConfigQueryParams 
    Request *shared.GoogleCloudRecommenderV1beta1RecommenderConfig `request:"mediaType=application/json"`
    Security RecommenderProjectsLocationsRecommendersUpdateConfigSecurity 
    
}

type RecommenderProjectsLocationsRecommendersUpdateConfigResponse struct {
    ContentType string 
    GoogleCloudRecommenderV1beta1RecommenderConfig *shared.GoogleCloudRecommenderV1beta1RecommenderConfig 
    StatusCode int64 
    
}

File: SDKs/googleapis.com/recommender/v1beta1/typescript/src/sdk/models/operations/recommenderprojectslocationsrecommendersupdateconfig.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";


export class RecommenderProjectsLocationsRecommendersUpdateConfigPathParams extends SpeakeasyBase {
  @Metadata({ data: "pathParam, style=simple;explode=false;name=name" })
  name: string;
}


export class RecommenderProjectsLocationsRecommendersUpdateConfigQueryParams extends SpeakeasyBase {
  @Metadata({ data: "queryParam, style=form;explode=true;name=$.xgafv" })
  dollarXgafv?: shared.XgafvEnum;

  @Metadata({ data: "queryParam, style=form;explode=true;name=access_token" })
  accessToken?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=alt" })
  alt?: shared.AltEnum;

  @Metadata({ data: "queryParam, style=form;explode=true;name=callback" })
  callback?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=fields" })
  fields?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=key" })
  key?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=oauth_token" })
  oauthToken?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=prettyPrint" })
  prettyPrint?: boolean;

  @Metadata({ data: "queryParam, style=form;explode=true;name=quotaUser" })
  quotaUser?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=updateMask" })
  updateMask?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=uploadType" })
  uploadType?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=upload_protocol" })
  uploadProtocol?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=validateOnly" })
  validateOnly?: boolean;
}


export class RecommenderProjectsLocationsRecommendersUpdateConfigSecurity extends SpeakeasyBase {
  @Metadata({ data: "security, scheme=true;type=oauth2" })
  oauth2: shared.SchemeOauth2;

  @Metadata({ data: "security, scheme=true;type=oauth2" })
  oauth2c: shared.SchemeOauth2c;
}


export class RecommenderProjectsLocationsRecommendersUpdateConfigRequest extends SpeakeasyBase {
  @Metadata()
  pathParams: RecommenderProjectsLocationsRecommendersUpdateConfigPathParams;

  @Metadata()
  queryParams: RecommenderProjectsLocationsRecommendersUpdateConfigQueryParams;

  @Metadata({ data: "request, media_type=application/json" })
  request?: shared.GoogleCloudRecommenderV1beta1RecommenderConfig;

  @Metadata()
  security: RecommenderProjectsLocationsRecommendersUpdateConfigSecurity;
}


export class RecommenderProjectsLocationsRecommendersUpdateConfigResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  googleCloudRecommenderV1beta1RecommenderConfig?: shared.GoogleCloudRecommenderV1beta1RecommenderConfig;

  @Metadata()
  statusCode: number;
}
